docs/reference/sql/data-definition.md

```diff
@@ -161,6 +161,53 @@ the key of the underlying {{ site.ak }} record, it must perform a repartition
 of the rows internally. If you're not sure what data is in the key or you simply
 don't need it, you can omit the `KEY` keyword.
 
+## Headers
+
+Starting in ksqlDB 0.24, you can mark a column with `HEADERS` or `HEADER('<key>')` to
+indicate that it is populated by the header field of the underlying {{ site.ak }} record.
+A column marked with `HEADERS` must have the type `ARRAY<STRUCT<key STRING, value BYTES>>`
+and contains the full list of the {{ site.ak }} record's header keys and values.
+
+A column marked with `HEADER('<key>')` must have the type `BYTES` and contains the last
+header that matches the key. If the {{ site.ak }} record does not contain a header with the
+specified key, then that column will be populated by `NULL`.
+
+In the following example statement, `k1`'s data is stored in the key portion of
+the row, `v1`'s data is stored in the value and `h1`'s data is stored in the header.
+
+```sql
+CREATE STREAM s4 (
+    k1 VARCHAR KEY,
+    v1 VARCHAR,
+    h1 BYTES HEADER('abc')
+) WITH (
+    kafka_topic = 's3',
+    value_format = 'json'
+);
+```
```

Comment:
Do you think it would be useful to provide an example query that uses `FROM_BYTES` to effectively read the header as a string?

Reply:
I added an example to that section. I'm not sure if there's another place in the docs that a more complete example could fit in.